From 4c2a971524e2f22d4c8e35b2bba1d109c43c9cc2 Mon Sep 17 00:00:00 2001
From: Vladimir Ivakin <vladimir_iva@pisem.net>
Date: Sun, 22 Mar 2015 15:37:17 +0300
Subject: [PATCH] remove libsystemd-daemon dependency

---
 Makefile.am  | 4 ++--
 configure.ac | 7 +++----
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index bf5553d..539284c 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -10,10 +10,10 @@ systemduserunit-install-hook:
 
 install-data-hook: systemduserunit-install-hook
 
-AM_CFLAGS = $(LIBSYSTEMD_DAEMON_CFLAGS) -Wall -Wno-uninitialized
+AM_CFLAGS = $(SYSTEMD_CFLAGS) -Wall -Wno-uninitialized
 AUTOMAKE_OPTIONS = subdir-objects
 
 bin_PROGRAMS = xorg-launch-helper
 xorg_launch_helper_SOURCES = src/main.c
-xorg_launch_helper_LDADD = $(LIBSYSTEMD_DAEMON_LIBS)
+xorg_launch_helper_LDADD = $(SYSTEMD_LIBS)
 
diff --git a/configure.ac b/configure.ac
index b2c71c9..c001721 100644
--- a/configure.ac
+++ b/configure.ac
@@ -16,10 +16,9 @@ AC_PROG_INSTALL
 AC_CHECK_LIB([rt], [main], ,
 	     AC_MSG_ERROR([librt is required but was not found]))
 
-PKG_CHECK_MODULES([SYSTEMD], [systemd])
-PKG_CHECK_MODULES([LIBSYSTEMD_DAEMON], [libsystemd-daemon])
-AC_SUBST(LIBSYSTEMD_DAEMON_CFLAGS)
-AC_SUBST(LIBSYSTEMD_DAEMON_LIBS)
+PKG_CHECK_MODULES([SYSTEMD], [libsystemd])
+AC_SUBST(SYSTEMD_CFLAGS)
+AC_SUBST(SYSTEMD_LIBS)
 
 AC_ARG_WITH([systemduserunitdir], AC_HELP_STRING([--with-systemduserunitdir=DIR],
 	[path to systemd user service directory]), [path_systemduserunit=${withval}],
-- 
2.2.1

