diff --git a/src/config/config.c b/src/config/config.c
index 7df4892..0b1b6ea 100644
--- a/src/config/config.c
+++ b/src/config/config.c
@@ -58,7 +58,7 @@ void cXVDRServerConfig::Load() {
   for (cSetupLine* l = First(); l; l = Next(l))
   {
     if(!Parse(l->Name(), l->Value()))
-      ERRORLOG("Unknown config parameter %s = %s in %s", l->Name(), l->Value(), GENERAL_CONFIG_FILE);
+      ERRORLOG("Unknown config parameter %s = %s in %s ", l->Name(), l->Value(), GENERAL_CONFIG_FILE);
   }
 
   cLiveQueue::RemoveTimeShiftFiles();
diff --git a/src/config/config.h b/src/config/config.h
index faa51ca..b25df7a 100644
--- a/src/config/config.h
+++ b/src/config/config.h
@@ -34,19 +34,19 @@
 // log output configuration
 
 #ifdef CONSOLEDEBUG
-#define DEBUGLOG(x...) printf("XVDR: "x)
+#define DEBUGLOG(x...) printf("XVDR: " x)
 #elif defined  DEBUG
-#define DEBUGLOG(x...) dsyslog("XVDR: "x)
+#define DEBUGLOG(x...) dsyslog("XVDR: " x)
 #else
 #define DEBUGLOG(x...)
 #endif
 
 #ifdef CONSOLEDEBUG
-#define INFOLOG(x...) printf("XVDR: "x)
-#define ERRORLOG(x...) printf("XVDR-Error: "x)
+#define INFOLOG(x...) printf("XVDR: " x)
+#define ERRORLOG(x...) printf("XVDR-Error: " x)
 #else
-#define INFOLOG(x...) isyslog("XVDR: "x)
-#define ERRORLOG(x...) esyslog("XVDR-Error: "x)
+#define INFOLOG(x...) isyslog("XVDR: " x)
+#define ERRORLOG(x...) esyslog("XVDR-Error: " x)
 #endif
 
 // default settings
diff --git a/src/live/livequeue.c b/src/live/livequeue.c
index bf27125..137739a 100644
--- a/src/live/livequeue.c
+++ b/src/live/livequeue.c
@@ -22,6 +22,7 @@
  *
  */
 
+#define __STL_CONFIG_H
 #include <sys/types.h>
 #include <dirent.h>
 #include <unistd.h>
diff --git a/src/xvdr/xvdrclient.c b/src/xvdr/xvdrclient.c
index ce249bc..a421e7a 100644
--- a/src/xvdr/xvdrclient.c
+++ b/src/xvdr/xvdrclient.c
@@ -23,6 +23,7 @@
  *
  */
 
+#define __STL_CONFIG_H
 #include <stdlib.h>
 #include <stdio.h>
 #include <sys/socket.h>
