--- a/common.h	2015-02-19 03:20:00.000000000 +0100
+++ b/common.h	2017-03-13 06:56:54.320000013 +0100
@@ -46,7 +46,7 @@
   do {                                                            \
      if (exp) {                                                   \
         char tmp[64];                                             \
-        error("[%s,%d]: "errstr": %s", __FILE__, __LINE__,        \
+        error("[%s,%d]: " errstr ": %s ", __FILE__, __LINE__,        \
               strerror_r(errno, tmp, sizeof(tmp)));               \
         func;                                                     \
         ret;                                                      \
